src/rpc/methods/eth.rs

```diff
@@ -1917,18 +2053,22 @@ impl RpcMethod<2> for EthGetTransactionCount {
         (sender, block_param): Self::Params,
     ) -> Result<Self::Ok, ServerError> {
         let addr = sender.to_filecoin_address()?;
-        let ts = tipset_by_block_number_or_hash(ctx.chain_store(), block_param)?;
+        if let BlockNumberOrHash::PredefinedBlock(ref predefined) = block_param {
+            if *predefined == Predefined::Pending {
+                return Ok(EthUint64(ctx.mpool.get_sequence(&addr)?));
+            }
+        }
+        let ts = tipset_by_block_number_or_hash(ctx.chain_store(), block_param.clone())?;
         let state = StateTree::new_from_root(ctx.store_owned(), ts.parent_state())?;
         let actor = state.get_required_actor(&addr)?;
         if is_evm_actor(&actor.code) {
             let evm_state = evm::State::load(ctx.store(), actor.code, actor.state)?;
             if !evm_state.is_alive() {
                 return Ok(EthUint64(0));
             }
-
             Ok(EthUint64(evm_state.nonce()))
         } else {
-            Ok(EthUint64(ctx.mpool.get_sequence(&addr)?))
+            Ok(EthUint64(actor.sequence))
```

Comment:
Is this a fix from previous code?

Reply:
this is changed to match logic as per lotus code https://github.com/filecoin-project/lotus/blob/97dc1865cf1e205ed5b5f18402a4d9e8cfa29141/node/impl/eth/transaction.go#L381